### PR TITLE
@@customeAzureResource decorator for resources that doesn't use the base resource types

### PR DESCRIPTION
**Pull request by AlitzelMendez**
issue: https://github.com/Azure/typespec-azure/issues/1474

### CODE REVIEW

File: packages/typespec-azure-resource-manager/lib/decorators.tsp [hunk 1]
```diff
@@ -182,6 +182,12 @@ extern dec armCommonTypesVersion(
  */
 extern dec armVirtualResource(target: Model);
 
+/**
+ * This decorator is used on resources that do not satisfy the definition of a resource
+ * but need to be identified as such.
+ */
+extern dec customAzureResource(target: Model);
```
Comment (markcowl) [inline packages/typespec-azure-resource-manager/lib/decorators.tsp:189]: I think we want this in the namespace `Azure.ResourceManager.Legacy`

File: packages/typespec-azure-resource-manager/src/lib.ts [hunk 2]
```diff
@@ -36,6 +36,12 @@ export const $lib = createTypeSpecLibrary({
         default: "Resource types must include a string property called 'name'.",
       },
     },
+    "arm-custom-resource-usage-discourage": {
+      severity: "warning",
+      messages: {
+        default: "Avoid using this decorator as it does not provide validation for ARM resources.",
```
Comment (markcowl) [inline packages/typespec-azure-resource-manager/src/lib.ts:42]:
```suggestion
        default: "Avoid using this decorator except when converting old APIs to TypeSpec, as it does not provide validation for ARM resources.",
```

File: packages/typespec-azure-resource-manager/src/rules/utils.ts [hunk 3]
```diff
@@ -93,6 +93,14 @@ export function isInternalTypeSpec(
   );
 }
 
+export function isLegacyTypeSpec(
```
Comment (markcowl) [inline packages/typespec-azure-resource-manager/src/rules/utils.ts:96]: I am not sure where this is used, but I would not expect Service TypeSpec to use `.Legacy` in their namespaces.

I think we should instead add a new linting rule that throws if folks use customAzureResource, and think about having a new 'legacy' ruleset that does not include this rule.  But I don't think the new ruleset needs to be in this PR

File: packages/typespec-azure-resource-manager/src/resource.ts [hunk 4]
```diff
@@ -92,6 +94,19 @@ export const $armVirtualResource: ArmVirtualResourceDecorator = (
   }
 };
 
+export const $customAzureResource: CustomAzureResourceDecorator = (
+  context: DecoratorContext,
+  entity: Model,
+) => {
+  const { program } = context;
+  if (isTemplateDeclaration(entity)) return;
+  if (!isLegacyTypeSpec(program, entity)) {
```
Comment (markcowl) [inline packages/typespec-azure-resource-manager/src/resource.ts:103]: I think we can just report this diagnostic.  Ideally, we would move this check into a linting rule, so we could eventually exclude it from a future 'arm brownfield'  ruleset

**Review by markcowl — CHANGES_REQUESTED** [review]

File: packages/typespec-azure-resource-manager/lib/decorators.tsp (hunk 1 above)
Comment (AlitzelMendez) [reply packages/typespec-azure-resource-manager/lib/decorators.tsp]: moved

File: packages/typespec-azure-resource-manager/src/rules/utils.ts (hunk 3 above)
Comment (AlitzelMendez) [reply packages/typespec-azure-resource-manager/src/rules/utils.ts]: removed this check

File: packages/typespec-azure-resource-manager/src/resource.ts (hunk 4 above)
Comment (AlitzelMendez) [reply packages/typespec-azure-resource-manager/src/resource.ts]: created the linting rule :)

**Review by markcowl — APPROVED** [review]

**Pull request merged**